packages/components/src/components/badge/badge.e2e-test.ts

```diff
@@ -1,22 +1,41 @@
-import { test } from '../../../config/playwright/setup';
+import { expect } from '@playwright/test';
+import { ComponentsPage, test } from '../../../config/playwright/setup';
+import { BADGE_TYPE, ICON_NAMES_LIST, DEFAULTS, ICON_VARIANT } from './badge.constants';
 
-test.beforeEach(async ({ componentsPage }) => {
+type SetupOptions = {
+  componentsPage: ComponentsPage;
+  type?: string;
+  iconName?: string;
+  variant?: string;
+  counter?: number;
+  maxCounter?: number;
+  ariaLabel?: string;
+};
+
+const setup = async (args: SetupOptions) => {
+  const { componentsPage, ...restArgs } = args;
   await componentsPage.mount({
     html: `
-        <mdc-badge />
+    <mdc-badge
+      ${restArgs.type ? `type="${restArgs.type}"` : ''}
+      ${restArgs.iconName ? `icon-name="${restArgs.iconName}"` : ''}
+      ${restArgs.variant ? `variant="${restArgs.variant}"` : ''}
+      ${restArgs.counter ? `counter="${restArgs.counter}"` : ''}
+      ${restArgs.maxCounter ? `max-counter="${restArgs.maxCounter}"` : ''}
+      ${restArgs.ariaLabel ? `ariaLabel="${restArgs.ariaLabel}"` : ''}
```

Comment:
HTML doesn't support camelCase attributes.
```suggestion
      ${restArgs.ariaLabel ? `aria-label="${restArgs.ariaLabel}"` : ''}
```